### PR TITLE
PHP 7 compatibility

### PR DESCRIPTION
**Pull request by thomask**
Cannot use 'String' as a class name since this is a reserved keyword in PHP 7. The conflicting class is not used anywhere by the bundle, so it was removed.

### CODE REVIEW

**Comment by thomask** [comment]
Fixes #196

**Comment by stof** [comment]
Removing the class is a BC break. Doing a new major version of the bundle just for that is not worth it.
This class is already deprecated, so keeping it inside the codebase does not make the codebase incompatible with PHP 7 (non-deprecated features are supported on PHP 7).

**Comment by thomask** [comment]
Ok agreed. How would you suggest handling the failing test case? (https://github.com/schmittjoh/JMSSecurityExtraBundle/blob/bbd5eb78da4d510478f1bb60959feefac083d474/Tests/Security/Util/StringTest.php#L11)

**Comment by Ener-Getick** [comment]
@thomask this test should be skipped.

File: composer.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
     ],
     "minimum-stability": "dev",
     "require": {
-        "php": "~5.3",
+        "php": ">=5.3",
```
Comment (Ener-Getick) [inline composer.json]: I prefer using ``~5.3|~7.0``

**Comment by thomask** [comment]
Closing as this will already be fixed in PR #202

**Pull request closed without merging**